### PR TITLE
 make generic function for xhr request

### LINKED ISSUE #18: make generic function for xhr requests

### PR DESCRIPTION
**Pull request by ahmadfarid-059**
Relates #18

- create explore folder 
- create fetch.js file 
- write a generic function for xhr request

### CODE REVIEW

File: public/api.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const fetchData = (url, cb) => {
+    const xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
```
Comment (asmaaamin) [inline public/api.js]: try to make it arrow function please

File: public/api.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+            if (xhr.status === 200) {
+                cb(JSON.parse(xhr.responseText));
+            } else {
+                cb(xhr.status);
+            }
+        }
+    };
+    xhr.open("GET", url);
+    xhr.send();
+}
```
Comment (umarsalah) [inline public/api.js]: Make a new folder called ```explorePage``` for the API request and initialize two script files one for ```fetch.js``` and another for the ```dom.js```, 
then in the fetch file after making the generic function , make a request for your API  in the same file .

File: public/api.js (hunk 1 above)
Comment (ahmadfarid-059) [reply public/api.js]: done

File: public/api.js (hunk 2 above)
Comment (ahmadfarid-059) [reply public/api.js]: done

**Review by asmaaamin — APPROVED** [review]

**Review by umarsalah — APPROVED** [review]

**Pull request merged**